src/azure-cli/azure/cli/command_modules/netappfiles/custom.py

```diff
@@ -476,7 +650,9 @@ def create_snapshot(cmd, client, resource_group_name, account_name, pool_name, v
                     location=None, no_wait=False):
     location = location or _get_location_from_resource_group(cmd.cli_ctx, resource_group_name)
     body = Snapshot(location=location)
-    return sdk_no_wait(no_wait, client.begin_create, resource_group_name, account_name, pool_name, volume_name, snapshot_name, body)
+    poller = client.begin_create(resource_group_name, account_name, pool_name, volume_name, snapshot_name, body)
+    LongRunningOperation(cmd.cli_ctx)(poller)
+    return sdk_no_wait(no_wait, client.get, resource_group_name, account_name, pool_name, volume_name, snapshot_name)
```

Comment:
The get operation is not LongRunningOperation, you should not use use sdk_no_wait for it.

Reply:
Agreed, thought that was wierd.  Will change